exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -48,18 +54,103 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 	if err != nil {
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
+	err = mgr.Add(&warmupRunnable{
```

Comment:
nit: can we add a note explaining why we need a runnable (to ensure extensions are discovered before any controllers start reconciling)